test/TestApps/ToastNotificationsTestApp/main.cpp

```diff
@@ -378,6 +378,51 @@ bool VerifyToastExpiresOnReboot()
     return true;
 }
 
+bool VerifyShowToast()
+{
+    // Registration happens in main()
+    winrt::ToastNotification toast{ GetToastNotification() };
+
+    auto toastNotificationManager = winrt::ToastNotificationManager::Default();
+    toastNotificationManager.ShowToast(toast);
+
+    if (toast.ToastId() == 0)
+    {
+        return false;
+    }
+
+    // TODO: Verify the toast was posted by calling History APIs.
+
+    return true;
+}
+
+bool VerifyShowToast_Unpackaged()
+{
+    winrt::ToastAssets assets(L"ToastNotificationApp", winrt::Uri{ LR"(C:\Windows\System32\WindowsSecurityIcon.png)" });
+    auto activationInfo = winrt::ToastActivationInfo::CreateFromToastAssets(assets);
+
+    auto toastNotificationManager = winrt::ToastNotificationManager::Default();
+    toastNotificationManager.RegisterActivator(activationInfo);
+
+    auto scope_exit = wil::scope_exit(
+        [&] {
+            toastNotificationManager.UnregisterActivator();
+        });
+
+    winrt::ToastNotification toast{ GetToastNotification() };
+
+    toastNotificationManager.ShowToast(toast);
```

Comment:
Are we certain that this test works? When I run it on my devbox, I don't see the toast, yet the test passes. If I run the packaged version of the test, I see the toast.

Reply:
Thanks for raising up this. We can't do more than this from a unit test perspective. We need E2E verification which I'm working on right now (manual verification).

Reply:
Actually, in the UDK tests we can have some form of E2E testing, if we implement the test toast sink. But this is not something we can get done before code complete. For now, we can rely on:

- Using History APIs after posting a toast
- Manual E2E verification

Reply:
I'm going to merge this change. Today I'll be tackling E2E testing. I will watch closely what happens with the unpackaged scenario. Any issues will be addressed in a future PR.